src/core/network.js

```diff
@@ -414,6 +415,17 @@ PDFNetworkStreamFullRequestReader.prototype = {
       networkManager.abortRequest(fullRequestXhrId);
     }
 
+    // Content-Disposition: attachment; filename=Naïve file.txt
+    var fullRequestXhr = networkManager.getRequestXhr(fullRequestXhrId);
+    var contentDisposition = fullRequestXhr
+                              .getResponseHeader('Content-Disposition');
+    if (contentDisposition) {
+      var parts = /.+; filename="(.+)"/gi.exec(contentDisposition);
```

Comment:
This patch also needs some unit tests in https://github.com/mozilla/pdf.js/blob/master/test/unit/network_spec.js. Since this line is the most essential part of the patch, I would suggest to perhaps extract these four lines into a separate function and test that in the unit test.